src/main/java/org/broadinstitute/dsm/model/at/DefaultValues.java

```diff
@@ -0,0 +1,63 @@
+package org.broadinstitute.dsm.model.at;
+
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.ParticipantData;
+import org.broadinstitute.dsm.db.dao.participant.data.ParticipantDataDao;
+import org.broadinstitute.dsm.model.participant.data.NewParticipantData;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class DefaultValues {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultValues.class);
+
+    private static final String FIELD_TYPE_ID = "AT_GROUP_GENOME_STUDY";
+    private static final String GENOME_STUDY_CPT_ID = "GENOME_STUDY_CPT_ID";
+    private static final String PREFIX = "DDP_ATCP_";
+
+    public static Map<String, List<ParticipantData>> addDefaultValues(Map<String, List<ParticipantData>> participantData,
+                                        Map<String, Map<String, Object>> participantESData, @NonNull DDPInstance instance, String queryAddition) {
+        if (participantESData == null) {
+            logger.warn("Could not update default values, participant ES data is null");
+            return participantData;
+        }
+        boolean addedNewParticipantData = false;
+        for (String ddpParticipantId : participantESData.keySet()) {
+            List<ParticipantData> participantDataList = participantData.get(ddpParticipantId);
+            //only needed for new signups - migrated pts already have the values needed
+            if (participantDataList == null) {
+                Map<String, Object> esParticipantData = participantESData.get(ddpParticipantId);
+                Map<String, Object> profile = (Map<String, Object>) esParticipantData.get(ElasticSearchUtil.PROFILE);
+                String hruid = (String) profile.get(ElasticSearchUtil.HRUID);
+                if (StringUtils.isNotBlank(hruid)) {
+                    ParticipantDataDao participantDataDao = new ParticipantDataDao();
```

Comment:
I would take create of participantDataDao out of for loop, just not to overload memory with create of many objects of ParticipantDataDao since they won't differ from each other

Reply:
good point